Source/DafnyLanguageServer/DafnyLanguageServer.cs

```diff
@@ -19,21 +21,51 @@ public static class DafnyLanguageServer {
       }
     }
 
-    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options, IConfiguration configuration) {
+    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options,
+        IConfiguration configuration, CancellationTokenSource cancelLanguageServer) {
       return options
         .WithDafnyLanguage(configuration)
         .WithDafnyWorkspace(configuration)
         .WithDafnyHandlers()
-        .OnInitialize(InitializeAsync)
+        .OnInitialize((server, @params, token) => InitializeAsync(server, @params, token, cancelLanguageServer))
         .OnStarted(StartedAsync);
     }
 
-    private static Task InitializeAsync(ILanguageServer server, InitializeParams request, CancellationToken cancellationToken) {
+    private static Task InitializeAsync(ILanguageServer server, InitializeParams request, CancellationToken cancelRequestToken,
+        CancellationTokenSource cancelLanguageServer) {
       var logger = server.GetRequiredService<ILogger<Program>>();
```

Comment:
That's something I must have missed when I extracted the `WithDafnyLanguageServer` method into a separate class. Could you change the injected logger instance to `ILogger<DafnyLanguageServer>`?